vms/platformvm/txs/fees/calculator.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fees
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/vms/platformvm/config"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+)
+
+var _ txs.Visitor = (*Calculator)(nil)
+
+type Calculator struct {
+	// Pre E-fork inputs
+	config    *config.Config
+	chainTime time.Time
```

Comment:
post E upgrade activation, we'll need to init different attributes (to be introduced in next PRs). I think it's cleaner to keep all the inputs private and use two different constructors to properly fill them.
Helps minimizing LOC changed